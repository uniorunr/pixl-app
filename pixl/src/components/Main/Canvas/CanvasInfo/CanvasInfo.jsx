import React from 'react';
import PropTypes from 'prop-types';
import './CanvasInfo.scss';

const CanvasInfo = ({ pixelsPerCanvas, currX, currY }) => (
  <div className="canvas-section__info">
    <span className="canvas-section__size">{`size: ${pixelsPerCanvas}x${pixelsPerCanvas}`}</span>
    <span className="canvas-section__position">
      {`x: ${currX >= 0 ? currX : 0}  y: ${currY >= 0 ? currY : 0}`}
    </span>
  </div>
);

CanvasInfo.propTypes = {
  pixelsPerCanvas: PropTypes.number.isRequired,
  currX: PropTypes.number.isRequired,
  currY: PropTypes.number.isRequired,
};

export default CanvasInfo;
