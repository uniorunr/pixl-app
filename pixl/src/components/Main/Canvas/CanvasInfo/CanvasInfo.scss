.canvas-section__info {
  margin: 5px 0;
  font-size: 1.2rem;
  font-weight: 700;
  width: 185px;
  display: flex;
  justify-content: space-between;

  .canvas-section__size {
    margin: 0 10px;
  }

  .canvas-section__position {
    margin: 0 10px;
  }
}
