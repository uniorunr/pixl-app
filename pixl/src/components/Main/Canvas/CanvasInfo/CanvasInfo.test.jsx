import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CanvasInfo from './CanvasInfo';
import store from '../../../../stores/mainStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <CanvasInfo />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
