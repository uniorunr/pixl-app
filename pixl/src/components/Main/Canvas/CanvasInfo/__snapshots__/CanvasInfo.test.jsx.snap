// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="canvas-section__info"
>
  <span
    className="canvas-section__size"
  >
    size: 64x64
  </span>
  <span
    className="canvas-section__position"
  >
    x: 0  y: 0
  </span>
</div>
`;
